src/brc20-bridge/src/ops.rs

```diff
@@ -0,0 +1,321 @@
+use bridge_canister::bridge::{Operation, OperationAction, OperationContext};
+use bridge_canister::runtime::RuntimeState;
+use bridge_did::error::{BftResult, Error};
+use bridge_did::op_id::OperationId;
+use bridge_did::order::{MintOrder, SignedMintOrder};
+use bridge_utils::bft_events::{
+    BurntEventData, MintedEventData, MinterNotificationType, NotifyMinterEventData,
+};
+use candid::{CandidType, Decode, Deserialize};
+use did::{H160, H256};
+use ic_exports::ic_cdk::api::management_canister::bitcoin::Utxo;
+use ic_task_scheduler::task::TaskOptions;
+use serde::Serialize;
+
+use crate::brc20_info::Brc20Tick;
+use crate::core::deposit::Brc20Deposit;
+
+#[derive(Debug, Serialize, Deserialize, CandidType, Clone)]
+pub enum Brc20BridgeOp {
+    // Deposit
+    AwaitInputs {
+        amount: u128,
+        brc20_tick: Brc20Tick,
+        dst_address: H160,
+        dst_token: H160,
+    },
+    AwaitConfirmations {
+        amount: u128,
+        brc20_tick: Brc20Tick,
+        dst_address: H160,
+        dst_token: H160,
+        utxos: Vec<Utxo>,
+    },
+    SignMintOrder {
+        dst_address: H160,
+        mint_order: MintOrder,
+    },
+    SendMintOrder {
+        dst_address: H160,
+        order: SignedMintOrder,
+    },
+    ConfirmMintOrder {
+        dst_address: H160,
+        order: SignedMintOrder,
+        tx_id: H256,
```

Comment:
Dst address can be queried from (Signed)MintOrder. So no need to store it explicitly.